byteps/common/common.h

```diff
@@ -250,6 +277,11 @@ ncclDataType_t getNcclDataType(DataType dtype);
 
 int getDataTypeLength(int dtype);
 
+inline size_t Align(size_t size, int dtype) {
+  const size_t min_size =
+      (getDataTypeLength(dtype) * getDataTypeLength(dtype)) * 8;
```

Comment:
Why multiply `getDataTypeLength(dtype)` twice?

Reply:
well, it does seem hard to understand. i should have made some comments here...

Aligning is for onebit algorithm. for example, for float32, data will be packing into int32. data size must be an integer multiple of 32*sizeof(int).  while 32 equals to sizeof(int) * 8. Then here comes the line282. This is the same for float16.

Reply:
btw, padding data is never used by any algorithms. for onebit, it eliminates annoying 'if-else` in the loop. but the padding part is not taken into account when calculating norm1 or topk or randomk.